mmdet/core/visualization/image.py

```diff
@@ -88,29 +125,37 @@ def imshow_det_bboxes(img,
         if segms is not None:
             segms = segms[inds, ...]
 
-    mask_colors = []
-    if labels.shape[0] > 0:
-        if mask_color is None:
-            # Get random state before set seed, and restore random state later.
-            # Prevent loss of randomness.
-            # See: https://github.com/open-mmlab/mmdetection/issues/5844
-            state = np.random.get_state()
-            # random color
-            np.random.seed(42)
-            mask_colors = [
-                np.random.randint(0, 256, (1, 3), dtype=np.uint8)
-                for _ in range(max(labels) + 1)
-            ]
-            np.random.set_state(state)
-        else:
-            # specify  color
-            mask_colors = [
-                np.array(mmcv.color_val(mask_color)[::-1], dtype=np.uint8)
-            ] * (
-                max(labels) + 1)
-
-    bbox_color = color_val_matplotlib(bbox_color)
-    text_color = color_val_matplotlib(text_color)
+    max_label = max(labels) if labels.shape[0] > 0 else -1
+
+    if (bbox_color is None) or (text_color is None) or (mask_color is None):
+        state = np.random.get_state()
+        # Get random state before set seed, and restore random state later.
+        # Prevent loss of randomness.
+        # See: https://github.com/open-mmlab/mmdetection/issues/5844
+        np.random.seed(42)
+        # random color
+        random_colors = [
+            np.random.randint(0, 256, (3, ), dtype=np.uint8)
+            for _ in range(max_label + 1)
+        ]
+        np.random.set_state(state)
+
+    bbox_color = random_colors if bbox_color is None else bbox_color
+    bbox_color = palette_val(bbox_color)
```

Comment:
move this line to below `if` part because get_palette() returns RGB format colors.